redfish-core/lib/power_supply_metrics.hpp

```diff
@@ -187,110 +189,199 @@ inline void
-                std::string,
-                std::vector<std::pair<std::string, std::vector<std::string>>>>>&
-                intfSubTree) mutable {
+            const std::variant<std::vector<std::string>>& endpoints) {
+            BMCWEB_LOG_DEBUG << "getValidInputHistory respHandler enter";
+
             if (ec)
             {
-                BMCWEB_LOG_DEBUG << "D-Bus response error on GetSubTree " << ec;
-                messages::internalError(aResp->res);
+                BMCWEB_LOG_ERROR
+                    << "getValidInputHistory respHandler D-Bus error: " << ec;
+                messages::internalError(asyncResp->res);
                 return;
             }
 
-            std::string serviceName;
-            std::string averagePath;
-            std::string maximumPath;
+            // Set the default value to resourceNotFound, and if we confirm
+            // finding association with chassisID and powerSupplyID is correct,
+            // the error response will be cleared.
+            messages::resourceNotFound(asyncResp->res, "PowerSupplyMetrics",
+                                       "Metrics");
+
+            const std::vector<std::string>* inputHistoryItem =
+                std::get_if<std::vector<std::string>>(&(endpoints));
 
-            for (const auto& [objectPath, connectionNames] : intfSubTree)
+            if (inputHistoryItem != nullptr)
             {
-                if (objectPath.empty())
+                // more
+                if ((*inputHistoryItem).size() == 0)
                 {
-                    BMCWEB_LOG_DEBUG << "Error getting D-Bus object!";
-                    messages::internalError(aResp->res);
+                    BMCWEB_LOG_ERROR
+                        << "Input history item association error! ";
+                    messages::internalError(asyncResp->res);
                     return;
                 }
 
-                const std::string& validPath = objectPath;
-                auto psuMatchStr = powerSupplyID + "_input_power";
-                std::string psuInputPowerStr;
-                // validPath -> {psu}_input_power
-                // 5 below comes from
-                // /org/open_power/sensors/aggregation/per_30s/
-                //   0      1         2         3         4
-                // .../{psu}_input_power/[average,maximum]
-                //           5..............6
-                if (!dbus::utility::getNthStringFromPath(validPath, 5,
-                                                         psuInputPowerStr))
-                {
-                    BMCWEB_LOG_ERROR << "Got invalid path " << validPath;
-                    messages::invalidObject(aResp->res, validPath);
-                    return;
-                }
+                // Clear resourceNotFound response
+                asyncResp->res.clear();
 
-                if (psuInputPowerStr != psuMatchStr)
+                std::vector<std::string> inputHistoryPath;
```

Comment:
Nit: Same as above.  Since this contains multiple paths, should it be named inputHistoryPaths with an 's'?

Reply:
It could be, but ... going with singular for now.